src/Bicep.Core/Emit/ResourceDependencyVisitor.cs

```diff
@@ -9,37 +9,53 @@
 
 namespace Bicep.Core.Emit
 {
+    /// <summary> Class <c>ResourceDependencyVisitor</c>  provides two public methods: <c>GetAllResourceDependencies, GetResourceDependencies</c>
```

Comment:
[nit] Avoid describing the code contract itself in comments - it's likely to get missed when refactors are made, and most IDEs make it simple to view this information by just looking at the code. Adding info that isn't part of the code contract is often worthwhile, but that probably makes more sense to go into comments on the methods themselves.